apps/ejabberd/src/mongoose_metrics.erl

```diff
@@ -194,14 +194,23 @@ get_odbc_stats(ODBCWorkers) ->
     PortStats = [inet_stats(Port) || Port <- lists:flatten(Ports)],
     [{workers, length(ODBCConnections)} | merge_stats(PortStats)].
 
-get_port_from_odbc_connection({{ok, DB, Pid}, WorkerPid}) when DB =:= mysql; DB =:= pgsql ->
+get_port_from_odbc_connection({{ok, DB, Pid}, WorkerPid}) when DB =:= mysql ->
     element(2, erlang:process_info(Pid, links)) -- [WorkerPid];
+get_port_from_odbc_connection({{ok, DB, Pid}, _WorkerPid}) when DB =:= pgsql ->
+    ProcStatus = sys:get_status(Pid),
+    get_port_from_proc_status(ProcStatus);
 get_port_from_odbc_connection({{ok, odbc, Pid}, WorkerPid}) ->
     Links = element(2, erlang:process_info(Pid, links)) -- [WorkerPid],
     [Port || Port <- Links, is_port(Port), {name, "tcp_inet"} == erlang:port_info(Port, name)];
 get_port_from_odbc_connection(_) ->
     undefined.
 
+get_port_from_proc_status({_, _, _, Status}) ->
```

Comment:
this function is awful, but i's fixed in another PR. so we can leave it as is here